src/wpool/mongoose_gun_worker.erl

```diff
@@ -0,0 +1,276 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% This module implements a wpool worker capable of establishing HTTP/2
+%%% connections using the `Gun' library.
+%%% @end
+%%%-------------------------------------------------------------------
+-module(mongoose_gun_worker).
+
+-behaviour(gen_server).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_continue/2,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2
+        ]).
+
+-record(gun_worker_state, {
+          host :: inet:hostname() | inet:ip_address(),
+          port :: inet:port_number(),
+          opts :: gun:opts(),
+          pid = undefined :: pid() | undefined,
```

Comment:
`undefined` is the default anyway. Is it for the sake of the explicitness? If so, it could be changed in the line below as well.